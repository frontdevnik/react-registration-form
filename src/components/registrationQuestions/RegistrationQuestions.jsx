import React from 'react';
import { useForm } from 'react-hook-form';
import { useSelector } from 'react-redux';

import {
  getFirstSelectAnswer,
  getSecondSelectAnswer,
  getThirdSelectAnswer,
} from '../../features/registrationQuestions/selectors';

import { setUpSelect } from '../../globals/materialize-elements-setting';

import MainSelect from '../mainSelect';

import {
  firstOptionText,
  firstOptions,
  secondOptions,
  secondOptionText,
  thirdOptions,
  thirdOptionText,
} from './defaultValues';

export default ({ onSubmit }) => {
  const question_one = useSelector(getFirstSelectAnswer);
  const question_two = useSelector(getSecondSelectAnswer);
  const question_three = useSelector(getThirdSelectAnswer);

  const { register, handleSubmit } = useForm();

  return (
    <form onSubmit={handleSubmit(onSubmit)} className="form">
      <MainSelect
        name="question_one"
        mainText={firstOptionText}
        options={firstOptions}
        register={setUpSelect(register)}
        value={question_one}
      />
      <MainSelect
        name="question_two"
        mainText={secondOptionText}
        options={secondOptions}
        register={setUpSelect(register)}
        value={question_two}
      />
      <MainSelect
        name="question_three"
        mainText={thirdOptionText}
        options={thirdOptions}
        register={setUpSelect(register)}
        value={question_three}
      />
      <button className="waves-effect waves-light btn">submit</button>
    </form>
  );
};
