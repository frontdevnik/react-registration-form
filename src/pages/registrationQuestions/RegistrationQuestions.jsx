import React from 'react';
import { useDispatch } from 'react-redux';

import MainWrapper from '../../components/mainWrapper';
import Breadcrumbs from '../../components/navigations';
import RegistrationQuestions from '../../components/registrationQuestions';

import {
  setRegistrationQuestionsField,
  submitRegistrationQuestionsForm,
} from '../../features/registrationQuestions/actions';

export default () => {
  const dispatch = useDispatch();

  const handleInput = ({ name, answer, questionText }) => {
    dispatch(setRegistrationQuestionsField({ name, answer, questionText }));
  };

  const handleSubmit = (event) => {
    event.preventDefault();

    dispatch(submitRegistrationQuestionsForm());
  };

  return (
    <MainWrapper>
      <RegistrationQuestions
        handleSubmit={handleSubmit}
        handleInput={handleInput}
      />
      <Breadcrumbs />
    </MainWrapper>
  );
};
