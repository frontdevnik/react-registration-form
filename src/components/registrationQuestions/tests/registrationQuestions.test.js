import React from 'react';

import RegistrationQuestions from '../RegistrationQuestions';

import { stubRegistrationQuestionsProps } from './stubProps';

jest.mock('react-redux', () => ({
  useSelector: jest.fn((f) => f()),
}));

jest.mock('react-hook-form', () => ({
  useForm: () => ({
    register: jest.fn(),
    handleSubmit: jest.fn((f) => f),
    reset: jest.fn(),
    errors: {},
  }),
}));

jest.mock('../../../features/registrationQuestions/selectors');

const setUp = (props) => render(<RegistrationQuestions {...props} />);

describe('<RegistrationQuestions />', () => {
  it('should render correctly', () => {
    const component = setUp(stubRegistrationQuestionsProps);
    expect(component.html()).toMatchSnapshot()
  });
});
