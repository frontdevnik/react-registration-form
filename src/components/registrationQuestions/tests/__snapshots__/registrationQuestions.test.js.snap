// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegistrationQuestions /> should render correctly 1`] = `"<div class=\\"input-field col s12\\"><select name=\\"question_one\\"><option value disabled selected>Choose your option</option><option value=\\"Male\\">Male</option><option value=\\"Female\\">Female</option><option value=\\"Refused\\">Refused</option></select><label>What is your gender?</label></div><div class=\\"input-field col s12\\"><select name=\\"question_two\\"><option value disabled selected>Choose your option</option><option value=\\"Morning\\">Morning</option><option value=\\"Afternoon\\">Afternoon</option><option value=\\"Evening\\">Evening</option></select><label>When do you eat chocolate?</label></div><div class=\\"input-field col s12\\"><select name=\\"question_three\\"><option value disabled selected>Choose your option</option><option value=\\"21\\">21</option><option value=\\"20\\">20</option></select><label>In which century you was born?</label></div><button class=\\"waves-effect waves-light btn\\">submit</button>"`;
